2025/incoming/the-gift-of-readability.pod

```diff
@@ -0,0 +1,450 @@
+tracking sleigh routes, Aurora Borealis hit ratios, and reindeer latency.
+
+All of it powered by Perl. Old Perl. New Perl.
+Perl written by elves long retired to toy-making or cloud-making.
+Perl written by skilled masters.
+Perl written by babies.
+
+She rubbed her eyes.
+
+Three hours. Three hours she'd been trying to understand this function.
+Three hours to fix what should have been a five-minute bug.
+
+The Present Delivery Network ran flawlessly — millions of gifts delivered every year with magical precision.
+But behind the scenes, the elves maintaining the codebase were drowning in their own cleverness.
+
+This code worked perfectly for ages, yet recently it stopped to feed Rudolph.
+And it is utterly incomprehensible for her.
+
+=begin code perl
+
+for my$r(@$rslts){push@${${$r->{feed}//next}->{$r->{reindeer}//next}},getPreferredReindeerFuel($r->{name}//next,$r->{prefs})if!!$r->{active}==!!1;}
+
+=end code
+
+=over
+
+=item * “Why,” she whispered to the empty office, “why do we do this to ourselves?”
+
+=item * "The Present Delivery Network was designed to be efficient."
+
+=item * "This code … less so."
+
+=back
+
+She thought about the astigmatic, colour-blind elf, squinting at compressed code through his magnifier.
+She thought about the foreigner elf, parsing English as their second language.
+She thought about the new elf starting after the holidays, terrified to touch anything.
+
+And so began the quiet refactoring ...
+
+=head2 The Invisible Barrier
+
+The Capability to Read is Not Given
+
+Readable code isn't just about aesthetics or dogma.
+
+It is about politeness.
+It is about inclusiveness.
+
+It is about respecting the time others have to spend dealing with our, ehm, work.
+Time spent in code review.
+Time lost while hunting bugs.
+Time newcomers need before they dare to change a line.
+
+We often forget that reading code is a learned skill layered on top of another learned skill: reading English.
+
+Whilst most programming languages and source code use English-based syntax,
+only about 400 million people speak English as their native language.
+Another about 1.5 billion speak it as a second language.
+Even among those who do, reading dense technical English is a different skill from ordering coffee or chatting about the weather.
+
+As an example, try to quickly read my favourite German word:
+
+=begin code
+
+Rindfleischetikettierungsüberwachungsaufgabenübertragungsgesetz
+
+=end code
+
+For those who do not speak German, it means:
+
+=over
+
+=item * The law for the delegation of duties for the supervision of the labelling of beef.
+
+=back
+
+Without delimiters, the brain struggles to find word boundaries. Code is no different.
+
+Machines execute instructions flawlessly; humans do not. Code is written once,
+but read many times, often by people who were not present when the first line was typed.
+
+=head2 How Humans Actually Read
+
+Children don’t read words as whole units. They decode letters, syllables, fragments.
```

Comment:
There is actually quite a bit of debate about this. Not that either side is right, but you're making a hard assertion on something that's a huge battle in elementary education, at least in the US.

I don't think you need any of this for the points you make though.